cluster/userdata-master.yaml

```diff
@@ -756,6 +760,36 @@ write_files:
               emptyDir:
                 medium: Memory # share a tmpfs between the two containers
 
+  - path: /srv/kubernetes/manifests/mate-de.yaml
+    content: |
+      apiVersion: extensions/v1beta1
+      kind: Deployment
+      metadata:
+        name: mate
+        namespace: kube-system
+      spec:
+        replicas: 1
+        template:
+          metadata:
+            labels:
+              app: mate
+          spec:
+            containers:
+            - name: mate
+              image: pierone.stups.zalan.do/teapot/mate:v0.0.4-31-gebb7747
+              args:
+              - --producer=kubernetes
+              - --kubernetes-domain=mate.teapot.zalan.do.
+              - --consumer=aws
+              - --aws-hosted-zone=mate.teapot.zalan.do.
+              - --no-sync
```

Comment:
mate knows about producers (source of records, e.g. kube services) and consumers (dns providers, e.g. route53)

it seems the zone is duplicated but that's not the full story. `--kubernetes-domain` is responsible for what dns records to create. `--aws-hosted-zone` just picks a zone on the aws side. (legacy from google)